crowbar_framework/app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class RepositoriesController < ApplicationController
+  before_filter :load_registry
+  #
+  # Repository Check
+  #
+  # Provides the restful api call for
+  # Repository Checks 	/utils/repositories 	GET 	Returns a json list of checked repositories
+  # Renders an HTML view in the UI
+  def index
+    @repocheck = Crowbar::Repository.check_all_repos
+    respond_to do |format|
+      format.html { @repocheck }
+      format.xml { render xml: @repocheck }
+      format.json { render json: @repocheck }
+    end
+  end
+
+  # update the state of the repositories (active/disabled)
+  # /utils/repositories/sync   POST
+  def sync
+    unless params["repo"].nil?
+      ProvisionerService.new(logger).synchronize_repositories(params["repo"])
+    end
+
+    redirect_to repositories_path
+  end
+
+  #
+  # Activate a single Repository
+  #
+  # Provides the restful api call for
+  # Activate a Repository   /utils/repositories/activate   POST  Creates Repository DataBagItem
+  # required parameters: platform, repo
+  def activate
+    return render_not_found if params[:platform].nil? || params[:repo].nil?
+    respond_to do |format|
+      repo_object = Crowbar::Repository.where(platform: params[:platform], repo: params[:repo]).first
+      ret = ProvisionerService.new(logger).create_repository_item(repo_object) unless repo_object.nil?
+      if ret
+        format.json { head :ok }
+        format.html { redirect_to repositories_url }
+      else
+        format.json do
+          render json: { error: I18n.t('cannot_activate_repo', scope: 'error', id: repo_object.id) },
+                 status: :unprocessable_entity
+        end
+        format.html do
+          flash[:alert] = I18n.t('cannot_activate_repo', scope: 'error', id: repo_object.id)
```

Comment:
Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping. (https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)